### ISSUE TITLE
performance: try flatparse

### ISSUE THREAD

**Issue by simonmichael** [simonmichael/hledger#1993]
Switching from megaparsec sounds like a large and risky project, but if it can provide similar error messages, could potentially bring a nice speedup in parsing (which is typically about half of our run time). Some experimentation and prototyping to check for feasibility would be useful.

https://github.com/AndrasKovacs/flatparse#features-and-non-features

**Comment by Xitian9** [comment]
This looks promising. I've experimented with improving parser performance, and this could certainly be an aspect of it. One caveat to note is this: **Only little-endian 64 bit systems are currently supported as the host machine.**

**Comment by simonmichael** [comment]
Indeed, I'm not sure what that rules out exactly but it sounds a bit limiting.
